types/xregexp/index.d.ts

```diff
@@ -83,7 +84,7 @@ declare namespace OuterXRegExp {
     function uninstall(options: Object): void;
     function uninstall(options: string): void;
 
-    function union(patterns: string[], flags?: string): RegExp;
+    function union(patterns: string[] | RegExp[], flags?: string): RegExp;
```

Comment:
Seems like this is actually `(string | RegExp)[]` - it need not be uniform. The docs show the example `XRegExp.union(['a+b*c', /(dogs)\1/, /(cats)\1/], 'i');`